pkg/exchange/coinbase/stream_handlers.go

```diff
@@ -85,50 +86,61 @@ func (s *Stream) handleConnect() {
 			logStream.Infof("bridge %s to level2 channel (%s)", sub.Channel, sub.Symbol)
-			subProductsMap["level2"] = append(subProductsMap["level2"], localSymbol)
+			subProductsMap[level2Channel] = append(subProductsMap[level2Channel], localSymbol)
 		case types.MarketTradeChannel:
+			// full/user channels both provide feeds on orders and trades
+			// full: all orders/trades on Coinbase Exchange
+			// user: orders/trades belong to the user
 			var localChannel types.Channel
 			if s.PublicOnly {
-				localChannel = "full"
+				localChannel = fullChannel
 			} else {
-				localChannel = "user"
+				localChannel = userChannel
 			}
 			subProductsMap[localChannel] = append(subProductsMap[localChannel], localSymbol)
 			logStream.Infof("bridge %s to %s(%s)", sub.Channel, localChannel, localSymbol)
+		case types.BookTickerChannel:
+			// ticker channel provides feeds on best bid/ask prices
+			subProductsMap[tickerChannel] = append(subProductsMap[tickerChannel], localSymbol)
+			logStream.Infof("bridge %s to %s(%s)", sub.Channel, tickerChannel, localSymbol)
 		case types.KLineChannel:
 			// TODO: add support to kline channel
-		case types.BookTickerChannel, types.AggTradeChannel, types.ForceOrderChannel, types.MarkPriceChannel, types.LiquidationOrderChannel, types.ContractInfoChannel:
-			logStream.Warnf("coinbase stream does not support subscription to %s", sub.Channel)
+		case types.AggTradeChannel, types.ForceOrderChannel, types.MarkPriceChannel, types.LiquidationOrderChannel, types.ContractInfoChannel:
+			logStream.Warnf("coinbase stream does not support subscription to %s, skipped", sub.Channel)
 		default:
+			// handle the channels that are not standard bbgo channels
 			if !s.bbgoChannelsOnly {
 				// rfqMatchChannel allow empty symbol
 				if sub.Channel != rfqMatchChannel && sub.Channel != statusChannel && len(sub.Symbol) == 0 {
 					continue
 				}
 				subProductsMap[sub.Channel] = append(subProductsMap[sub.Channel], localSymbol)
 			} else {
-				logStream.Warnf("do not support subscription to non-standard channel %s (use `.StandardStream.Subscribe` to bypass it)", sub.Channel)
+				logStream.Warnf("subscription to non-standard channel %s disabled (use `.SetBbgoChannelsOnly(false)` to enable it)", sub.Channel)
 			}
 		}
 	}
 
-	_, subscribeFull := subProductsMap[fullChannel]
-	s.userOrderOnly = !subscribeFull
+	s.subLocalChannelsMap = make(map[types.Channel]struct{})
+	for channel := range subProductsMap {
+		s.subLocalChannelsMap[channel] = struct{}{}
+	}
 
+	// do subscription
 	var subCmds []any
 	signature, ts := s.generateSignature()
 	for channel, productIDs := range subProductsMap {
 		var subType string
-		switch types.Channel(channel) {
+		switch channel {
 		case rfqMatchChannel:
 			subType = "subscriptions"
 		default:
 			subType = "subscribe"
 		}
 		var subCmd any
-		switch types.Channel(channel) {
+		switch channel {
 		case statusChannel:
 			subCmd = subscribeMsgType1{
 				Type: subType,
```

Comment:
Does https://github.com/c9s/bbgo/pull/1945/commits/4dc1d4e003899c97b9332d71e96988c43f8ec32d#diff-ff2579743884f3142934a2212709fd2eb1c2e129710e57ecd0b8ed2a1c219332R200 require authentication?

Reply:
I handle the authentication [here](https://github.com/c9s/bbgo/blob/cbcb027cf7319905c12c5bd15cb019a4c0988ea9/pkg/exchange/coinbase/stream_handlers.go#L134-L245).